traits/src/resource.rs

```diff
@@ -108,57 +108,49 @@ pub enum ResourceTypes<BoundedString, BoundedParts> {
 	Slot(SlotResource<BoundedString>),
 }
 
-
 #[derive(Encode, Decode, Eq, PartialEq, Clone, RuntimeDebug, TypeInfo, MaxEncodedLen)]
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
-pub struct ResourceInfo<BoundedResource, BoundedString, BoundedParts> {
```

Comment:
Couldn't we keep `src`, `metadata`, `license` & `thumb` in here? I was thinking `ResourceInfo` would have the `ResourceType` that we could `match` on then get the contents from the nested struct then handle logic from there.